sdk/nodejs/runtime/rpc.ts

```diff
@@ -578,54 +586,113 @@ export function suppressUnhandledGrpcRejections<T>(p: Promise<T>): Promise<T> {
     return p;
 }
 
+function sameVersion(a?: semver.SemVer, b?: semver.SemVer): boolean {
+    // We treat undefined as a wildcard, so it always equals every other version.
+    return a === undefined || b === undefined || semver.eq(a, b);
+}
+
+function checkVersion(want?: semver.SemVer, have?: semver.SemVer): boolean {
+    if (want === undefined || have === undefined) {
+        return true;
+    }
+    return have.major === want.major && have.minor >= want.minor && have.patch >= want.patch;
+}
+
 /**
  * A ResourcePackage is a type that understands how to construct resource providers given a name, type, args, and URN.
  */
 export interface ResourcePackage {
-    constructProvider(name: string, type: string, args: any, opts: { urn: string }): ProviderResource;
+    readonly version: semver.SemVer | undefined;
```

Comment:
Should we keep the return type `string` to avoid the need for packages to depend on the `semver` package?

Reply:
Yeah, I'm going to change this to be a string -- otherwise, we can run into TypeScript errors if we don't include `@types/semver` as a runtime dependency.